go/teams/member_test.go

```diff
@@ -262,26 +264,41 @@ func TestMemberRemove(t *testing.T) {
 	assertRole(tc, name, other.Username, keybase1.TeamRole_NONE)
 
 	teamID, seqno := pollForNextMerkleRootAfterRemovalViaLibkb(t, tc, other, name)
-	testFindNextMerkleRootAfterRemoval(t, tc, other, teamID, seqno)
+	testFindNextMerkleRootAfterRemoval(t, tc, other, teamID, anyRoleAllowed, seqno)
 }
 
-func TestMemberChangeRole(t *testing.T) {
+// Set up log points for a user in a team with roles of
+// Writer->Reader->Writer->Reader and verify that the first merkle root
+// after the last demotion from writer points to the last sequence number.
+func TestMemberRemoveWriter(t *testing.T) {
 	tc, owner, other, _, name := memberSetupMultiple(t)
 	defer tc.Cleanup()
 
+	anyRoleAllowed := false
 	if err := SetRoleWriter(context.TODO(), tc.G, name, other.Username); err != nil {
 		t.Fatal(err)
 	}
-
 	assertRole(tc, name, owner.Username, keybase1.TeamRole_OWNER)
 	assertRole(tc, name, other.Username, keybase1.TeamRole_WRITER)
 
 	if err := SetRoleReader(context.TODO(), tc.G, name, other.Username); err != nil {
 		t.Fatal(err)
 	}
+	assertRole(tc, name, other.Username, keybase1.TeamRole_READER)
+
+	if err := SetRoleWriter(context.TODO(), tc.G, name, other.Username); err != nil {
+		t.Fatal(err)
+	}
+	assertRole(tc, name, other.Username, keybase1.TeamRole_WRITER)
 
+	if err := SetRoleReader(context.TODO(), tc.G, name, other.Username); err != nil {
+		t.Fatal(err)
+	}
 	assertRole(tc, name, owner.Username, keybase1.TeamRole_OWNER)
 	assertRole(tc, name, other.Username, keybase1.TeamRole_READER)
+
+	teamID, seqno := pollForNextMerkleRootAfterRemovalViaLibkb(t, tc, other, name)
+	testFindNextMerkleRootAfterRemoval(t, tc, other, teamID, anyRoleAllowed, seqno)
```

Comment:
Is there any way to test that you're getting the last demotion and not the first demotion?

Reply:
yeah - is this not doing that? `pollForNextMerkleRootAfterRemovalViaLibkb` essentially just gets the current sequence number.

Reply:
ok great, so this test failed before the `break` was inserted into the loop?

Reply:
yes.

Reply:
👍